File: src/main/java/com/github/eendroroy/sdk/openweather/response/GetWeatherResponse.java
package com.github.eendroroy.sdk.openweather.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonRootName;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import java.util.List;

/**
 * @author indrajit
 */

@ToString
@Setter
@Getter
@JsonRootName("weather")
public class GetWeatherResponse implements BaseResponse {
    @JsonProperty("coord")
    public CoOrdResponse coOrd;

    @JsonProperty("weather")
    public List<WeatherResponse> weather = null;

    @JsonProperty("base")
    public String base;

    @JsonProperty("main")
    public MainResponse main;

    @JsonProperty("visibility")
    public Integer visibility;

    @JsonProperty("wind")
    public WindResponse wind;

    @JsonProperty("clouds")
    public CloudsResponse clouds;

    @JsonProperty("dt")
    public Integer dt;

    @JsonProperty("sys")
    public SysResponse sys;

    @JsonProperty("timezone")
    public Integer timezone;

    @JsonProperty("id")
    public Integer id;

    @JsonProperty("name")
    public String name;

    @JsonProperty("cod")
    public Integer cod;

    @JsonProperty("message")
    public String message;
}

File: src/main/kotlin/com/github/eendroroy/sdk/openweathermap/response/GetWeatherResponse.kt
package com.github.eendroroy.sdk.openweathermap.response

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonRootName
import com.github.eendroroy.sdk.openweathermap.response.embedded.*

/**
 * @author indrajit
 */
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonRootName("weather")
data class GetWeatherResponse(
        @JsonProperty("coord")      var coOrd: CoOrd?,
        @JsonProperty("base")       var base: String?,
        @JsonProperty("main")       var main: Main?,
        @JsonProperty("visibility") var visibility: Int?,
        @JsonProperty("wind")       var wind: Wind?,
        @JsonProperty("clouds")     var clouds: Clouds?,
        @JsonProperty("dt")         var dt: Int?,
        @JsonProperty("sys")        var sys: Sys?,
        @JsonProperty("timezone")   var timezone: Int?,
        @JsonProperty("id")         var id: Int?,
        @JsonProperty("name")       var name: String?,
        @JsonProperty("cod")        var cod: Int?,
        @JsonProperty("message")    var message: String?,
        @JsonProperty("weather")    var weather: List<Weather>?
) : BaseResponse {
    override var httpCode: String? = null
    override var httpMessage: String? = null
}